docs/tasks/manage-gpus/scheduling-gpus.md

```diff
@@ -4,154 +4,203 @@ approvers:
 
 ```yaml
-kind: pod
 apiVersion: v1
+kind: Pod
 metadata:
-  annotations:
-    scheduler.alpha.kubernetes.io/affinity: >
-      {
-        "nodeAffinity": {
-          "requiredDuringSchedulingIgnoredDuringExecution": {
-            "nodeSelectorTerms": [
-              {
-                "matchExpressions": [
-                  {
-                    "key": "alpha.kubernetes.io/nvidia-gpu-name",
-                    "operator": "In",
-                    "values": ["Tesla K80", "Tesla P100"]
-                  }
-                ]
-              }
-            ]
-          }
-        }
-      }
+  name: cuda-vector-add
 spec:
+  restartPolicy: OnFailure
   containers:
-    -
-      name: gpu-container-1
+    - name: cuda-vector-add
+      image: "gcr.io/google_containers/cuda-vector-add:v0.1"
       resources:
         limits:
-          alpha.kubernetes.io/nvidia-gpu: 2
+          nvidia.com/gpu: 1 # requesting 1 GPU
+```
+
+### Deploying NVIDIA GPU device plugin
+
+There are currently two device plugin implementations for NVIDIA GPUs:
+
+#### Official NVIDIA GPU device plugin
+
+The [official NVIDIA GPU device plugin](https://github.com/NVIDIA/k8s-device-plugin)
+has the following requirements:
+- Kubernetes nodes have to be pre-installed with NVIDIA drivers.
+- Kubernetes nodes have to be pre-installed with [nvidia-docker 2.0](https://github.com/NVIDIA/nvidia-docker)
+- nvidia-container-runtime configured as the [default runtime](https://github.com/NVIDIA/nvidia-docker/wiki/Advanced-topics#default-runtime)
+  for docker instead of runc.
+- NVIDIA drivers ~= 361.93
+
+To deploy the NVIDIA device plugin once your cluster is running and above
+requirements are satisfied:
+
+```
+# For Kubernetes v1.8
+kubectl create -f https://raw.githubusercontent.com/NVIDIA/k8s-device-plugin/v1.8/nvidia-device-plugin.yml
+
+# For Kubernetes v1.9
+kubectl create -f https://raw.githubusercontent.com/NVIDIA/k8s-device-plugin/v1.9/nvidia-device-plugin.yml
+```
+
+#### NVIDIA GPU device plugin used by GKE/GCE
+
+The [NVIDIA GPU device plugin used by GKE/GCE](https://github.com/GoogleCloudPlatform/container-engine-accelerators/tree/master/cmd/nvidia_gpu)
+doesn't require using nvidia-docker and should work with any CRI compatible container
+runtime. It's supported on [COS](https://cloud.google.com/container-optimized-os/)
+and has experimental support for Ubuntu from 1.9 onwards.
+
+On your 1.9 cluster, you can use the following commands to install NVIDIA drivers and device plugin:
+
 ```
+# Install NVIDIA drivers on COS:
+kubectl create -f https://raw.githubusercontent.com/GoogleCloudPlatform/container-engine-accelerators/k8s-1.9/daemonset.yaml
 
-This will ensure that the pod will be scheduled to a node that has a `Tesla K80` or a `Tesla P100` Nvidia GPU.
+# Install NVIDIA drivers on Ubuntu:
+kubectl create -f https://raw.githubusercontent.com/GoogleCloudPlatform/container-engine-accelerators/k8s-1.9/nvidia-driver-installer/ubuntu/daemonset.yaml
```

Comment:
Please add that NVIDIA does not officially support installing drivers from a container on Ubuntu and that any support will be handled by the GKE team

Reply:
This is OSS documentation. I don't think there's any expectation of support without buying a support package from some vendor.

In any case, I added experimental at the top.

Reply:
OSS doesn't mean people won't open issues on the nvidia-docker repository or the nvidia k8s-device plugin.

If you are going to do and advertise something that is not supported by NVIDIA (because there are know bugs and that this going to break in a lot of edge cases due to driver issues) at the very least you could say that it's not supported by NVIDIA.

Reply:
Added directions to report issues with each device plugin.